clang_delta/EmptyStructToInt.cpp

```diff
@@ -213,6 +261,58 @@ void EmptyStructToInt::HandleTranslationUnit(ASTContext &Ctx)
     TransError = TransInternalError;
 }
 
+void EmptyStructToInt::handleOneRecordDecl(const RecordDecl *RD,
+                                           const RecordDecl *BaseRD,
+                                           const FieldDecl *FD,
+                                           unsigned int Idx)
+{
+    // Skip field if it is not ofthe type of one of the structs that are affected by the change
+    // Works recursively through chained structs
+    if(!RecordDeclToField[BaseRD] && BaseRD != TheRecordDecl)
+    {
+        return;
+    }
+
+    IndexVector *NewIdxVec = RecordDeclToField[RD];
+
+    if(!NewIdxVec)
+    {
+        NewIdxVec = new IndexVector();
+        RecordDeclToField[RD] = NewIdxVec;
+    }
+
+    NewIdxVec->push_back(Idx);
+}
+
+void EmptyStructToInt::handleOneVarDecl(const VarDecl *VD)
+{
```

Comment:
Since we are dealing with ```structs``` that have at most one field. Can we simplify the logic to something like:

    const Expr *IE = VD->getInit();
    if (!IE)
      return true;
    const Type *Ty = VD->getType().getTypePtr();
    const RecordDecl *RD = getBaseRecordDef(Ty);
    if (RD->getCanonicalDecl() == TheRecordDecl)
      RewriteHelper->replaceExpr(IE, "0")
    return true;